ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -815,6 +841,9 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
             then Nothing
             else Just $ LoEEnabled $ k + loeSuffixLength - rollback
         else
+          -- REVIEW: Since we filter out candidates that fork off the LoE later,
+          -- it does not really seem to matter what we return here and we could
+          -- maybe just return @Nothing@ or merge it in the previous case?
           if rollback > k
             then Nothing
             else Just $ LoEEnabled $ k - rollback
```

Comment:
Returning `Nothing` causes the current selection to be left unmodified. I don't think this is what we would want.

The actual value of `k - rollback` is used to decide how long to build the candidates that don't extend the LoE. I like that `computeLoEMaxExtra` returns something meaningful in this scenario.

If less candidates or shorter candidates need to be produced, changes would need to be done elsewhere.

Reply:
I am not sure that the value within `Just` makes a real difference in this scenario, however I can believe my proposed change makes a difference in chain selection by transforming a `Just` into a nothing, when we use that to decide in which case we find ourselves.

Reply:
`computeLoEMaxExtra` is now gone